### PR TITLE
[JBPM-8885] Nodes after multi-Instance subprocess are not blurred eventhough process instance is completed.

### PR DESCRIPTION
**Pull request by elguardian**
Added a property to increase the number of nodes retrieved from database

### CODE REVIEW

**Comment by elguardian** [comment]
Reuse of queries cannot happens as it will have side effects in other places of the code
Cannot create custom queries for retrieve just the ids set of the nodes (the image process also depends on the identity column)
just left with a custom property and increase the number so I don't hurt too much the performance.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java [hunk 1]
```diff
@@ -41,11 +41,19 @@
...
 public class ImageServiceBase {
 
     private static final Logger logger = LoggerFactory.getLogger(ImageServiceBase.class);
 
+    /**
+     * This causes the image service to limit the number of nodes (performance). 
+     * Due to this limitation it could cause a known issue not blurring all the nodes active or completed depending on
+     * the process size.
+     */
+    private static final int MAX_NODES = Integer.parseInt(System.getProperty(KIE_SERVER_IMAGESERVICE_MAX_NODES, "10000"));
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: I would make the default `1000` as it was before. Just for the sake of backward compatibility.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java [hunk 2]
```diff
@@ -121,8 +129,8 @@ public String getActiveProcessImage(String containerId, long procInstId, String
         if (imageSVG != null) {
             // find active nodes and modify image
             Map<String, String> subProcessLinks = new HashMap<>();
-            Collection<NodeInstanceDesc> activeLogs = dataService.getProcessInstanceHistoryActive(procInstId, new QueryContext(0, 1000));
-            Collection<NodeInstanceDesc> completedLogs = dataService.getProcessInstanceHistoryCompleted(procInstId, new QueryContext(0, 1000));
+            Collection<NodeInstanceDesc> activeLogs = dataService.getProcessInstanceHistoryActive(procInstId, new QueryContext(0, MAX_NODES));
+            Collection<NodeInstanceDesc> completedLogs = dataService.getProcessInstanceHistoryCompleted(procInstId, new QueryContext(0, MAX_NODES));
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: Did you consider using paging for that?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java (hunk 1 above)
Comment (MarianMacik) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: OK, if we take it as a fix, I see no problem. But the behaviour changes slightly now :)

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java (hunk 2 above)
Comment (MarianMacik) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: Can you please elaborate on this?
> yep but if the limit (X) there is for performance reason it does not make much sense to retrieve in chunks. you do it all in one go (just need to pass query context as null) .

Not sure I got it.

But yes, we have to fetch all data at once, I was mistaken by the fact that we are doing some checks for it or we are showing progress, but neither of that is true. :)

Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: I don't want to remove the limit (there must be a reason), but I will allow to retrieve everything if the parameter is 0 (infinity).

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: I meant I will change it.

Comment (MarianMacik) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: OK then. Thanks.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java (hunk 2 above)
Comment (MarianMacik) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java]: Thanks, that makes sense.

**Comment by MarianMacik** [comment]
@mswiderski any comments?

**Pull request merged**